File: sdkwork-ai-sdk-swift/Sources/API/Embeddings2Api.swift
import Foundation

public class Embeddings2Api {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Create embeddings
    public func createEmbedding(body: EmbeddingRequest) async throws -> EmbeddingResponse? {
        let response = try await client.post(ApiPaths.aiPath("/embeddings"), body: body)
        return response as? EmbeddingResponse
    }
}

File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/api/Embeddings2Api.kt
package com.sdkwork.ai.api

import com.sdkwork.ai.*
import com.sdkwork.ai.http.HttpClient

class Embeddings2Api(private val client: HttpClient) {

    /** Create embeddings */
    suspend fun createEmbedding(body: EmbeddingRequest): EmbeddingResponse? {
        return client.post(ApiPaths.aiPath("/embeddings"), body) as? EmbeddingResponse
    }
}
